app/javascript/src/components/Editor.tsx

```diff
@@ -100,6 +100,10 @@ class Editor extends React.Component<EditorProps, EditorState> {
     let currentState = clone(this.serialize())
     let pastState = this.past[this.past.length - 1]
 
+    if(currentState.story.links > pastState.story.links) {
+      this.handleSceneName(currentState, newStory)
```

Comment:
The main reason for putting this logic here is because currentState is actively grabbing the diagram data as it happens. I found that this would be the most consistent state to check against when deciding whether to run this logic or not

Reply:
I think this is worth a little comment describing the motivation here.

Reply:
When testing, this seemed to be a pretty solid check to give me the proper information to know whether to run the logic to check if the name needs to be updated. I tried testing it a bunch and didn't come across any side effects. Do you think this approach is sustainable?

Reply:
Ah yeah, it looks sustainable. I'd recommend a comment above the code though explaining to future devs that this is a slightly unusual way to tap into the event that the user added a new link.

Reply:
Yeah.. good idea. Do you think as a 2nd pass to this implementation it might benefit from making some kind of event handler/callback for detecting when a new link has been made? I feel like that should have been in the diagram library but I was unsuccessful in finding one.